x/subscription/types/message_subscribe.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgSubscribe = "subscribe"
+
+var _ sdk.Msg = &MsgSubscribe{}
+
+func NewMsgSubscribe(creator string, consumer string, index string, isYearly bool) *MsgSubscribe {
+	return &MsgSubscribe{
+		Creator:  creator,
+		Consumer: consumer,
+		Index:    index,
+		IsYearly: isYearly,
+	}
+}
+
+func (msg *MsgSubscribe) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgSubscribe) Type() string {
+	return TypeMsgSubscribe
+}
+
+func (msg *MsgSubscribe) GetSigners() []sdk.AccAddress {
```

Comment:
why do you return an array if there's always one creator?

Reply:
this is scaffolded function